packages/common/src/utils/filters.ts

```diff
@@ -689,16 +690,47 @@ export const overrideChartFilter = (
               ),
           };
 
+const getDeduplicatedFilterRules = (
+    filterRules: FilterRule[],
+    filterGroup?: FilterGroup,
+): FilterRule[] => {
+    const groupFilterRules = getFilterRulesFromGroup(filterGroup);
+    return filterRules.filter(
+        (rule) =>
+            !groupFilterRules.some((groupRule) => groupRule.id === rule.id),
+    );
+};
+
 const overrideFilterGroupWithFilterRules = (
     filterGroup: FilterGroup | undefined,
     filterRules: FilterRule[],
-): FilterGroup => ({
-    id: uuidv4(),
-    and: [
-        ...(filterGroup ? [overrideChartFilter(filterGroup, filterRules)] : []),
-        ...filterRules,
-    ],
-});
+): FilterGroup => {
+    if (!filterGroup) {
+        return {
+            id: uuidv4(),
+            and: filterRules,
+        };
+    }
+
+    const overriddenGroup = overrideChartFilter(filterGroup, filterRules);
+
+    // deduplicate the dashboard filter rules from the ones used when overriding the chart filterGroup
```

Comment:
I guess this is ok because we recently said chart filters should NOT be used for limiting data access...